### PR TITLE
[Adagrad optimization] adding initial_accumulator_value parameter to Adagrad

### PR DESCRIPTION
**Pull request by atulkum**
adding initial_accumulator_value parameter to Adagrad optimization algorithm

### CODE REVIEW

**Comment by SsnL** [comment]
Why is this option useful?

**Comment by atulkum** [comment]
It was needed for implementing the paper "[Get To The Point: Summarization with Pointer-Generator Networks] (https://github.com/abisee/pointer-generator) ". I was looking for equivalent pytorch implementation for Tensorflow AdagradOptimizer(https://www.tensorflow.org/api_docs/python/tf/train/AdagradOptimizer).

What I understood is that the effective learning rate "lr/sqrt(G_ii + eps)" would be very high in the initial phase of optimization if G_ii starts from 0. So to make the loss fluctuation smooth, its better to initialize G_ii with some small value.  (reference for notation: http://ruder.io/optimizing-gradient-descent/index.html#adagrad)

File: torch/optim/adagrad.py [hunk 1]
```diff
@@ -19,22 +19,25 @@ class Adagrad(Optimizer):
...
+        defaults = dict(lr=lr, lr_decay=lr_decay, weight_decay=weight_decay,
+                        initial_accumulator_value=initial_accumulator_value)
         super(Adagrad, self).__init__(params, defaults)
 
         for group in self.param_groups:
             for p in group['params']:
                 state = self.state[p]
                 state['step'] = 0
-                state['sum'] = torch.zeros_like(p.data)
+                state['sum'] = torch.zeros_like(p.data).fill_(initial_accumulator_value)
```
Comment (apaszke) [inline torch/optim/adagrad.py]: Can you please change it to `empty_like`? There's no point in filling it with zeros if it's going to be overwritten anyway

**Review by apaszke — COMMENTED** [review]
LGTM, but needs a single fix before merge

File: torch/optim/adagrad.py (hunk 1 above)
Comment (SsnL) [reply torch/optim/adagrad.py]: Or better, use 
```python
torch.full_like(p, initial_accumulator_value)
```

Comment (SsnL) [reply torch/optim/adagrad.py]: we should really document these new methods better...

**Review by apaszke — APPROVED** [review]
The change was minor so I did it for you.
